tests/gradients/test_parameter_shift_cv.py

```diff
@@ -0,0 +1,1105 @@
+            # entangle the modes
+            qml.Beamsplitter(1.0, 0.0, wires=[0, 1])
+            # Non-Gaussian
+            qml.Kerr(1.0, wires=[1])
+            qml.expval(qml.P(0))
+            qml.expval(qml.X(1))
+
+        # After entangling the modes, the Kerr gate now succeeds
+        # both initial rotations
+        assert _grad_method(tape, 0) == "F"
+        assert _grad_method(tape, 1) == "F"
+        assert _grad_method(tape, 2) == "F"
+
+    def test_probability(self):
+        """Probability is the expectation value of a
+        higher order observable, and thus only supports numerical
+        differentiation"""
+        with qml.tape.JacobianTape() as tape:
+            qml.Rotation(0.543, wires=[0])
+            qml.Squeezing(0.543, 0, wires=[0])
+            qml.probs(wires=0)
+
+        assert _grad_method(tape, 0) == "F"
+        assert _grad_method(tape, 1) == "F"
+        assert _grad_method(tape, 2) == "F"
+
+    def test_variance(self):
+        """If the variance of the observable is first order, then
+        parameter-shift is supported. If the observable is second order,
+        however, only finite-differences is supported."""
+
+        with qml.tape.JacobianTape() as tape:
+            qml.Rotation(1.0, wires=[0])
+            qml.var(qml.P(0))  # first order
+
+        assert _grad_method(tape, 0) == "A"
+
+        with qml.tape.JacobianTape() as tape:
+            qml.Rotation(1.0, wires=[0])
+            qml.var(qml.NumberOperator(0))  # second order
+
+        assert _grad_method(tape, 0) == "F"
+
+        with qml.tape.JacobianTape() as tape:
+            qml.Rotation(1.0, wires=[0])
+            qml.Rotation(1.0, wires=[1])
+            qml.Beamsplitter(0.5, 0.0, wires=[0, 1])
+            qml.var(qml.NumberOperator(0))  # second order
+            qml.expval(qml.NumberOperator(1))
+
+        assert _grad_method(tape, 0) == "F"
+        assert _grad_method(tape, 1) == "F"
+        assert _grad_method(tape, 2) == "F"
+        assert _grad_method(tape, 3) == "F"
+
+    def test_second_order_expectation(self):
+        """Test that the expectation of a second-order observable forces
+        the gradient method to use the second-order parameter-shift rule"""
+
+        with qml.tape.JacobianTape() as tape:
+            qml.Rotation(1.0, wires=[0])
+            qml.expval(qml.NumberOperator(0))  # second order
+
+        assert _grad_method(tape, 0) == "A2"
+
+    def test_unknown_op_grad_method(self, monkeypatch):
+        """Test that an exception is raised if an operator has a
+        grad method defined that the CV parameter-shift tape
+        doesn't recognize"""
+        monkeypatch.setattr(qml.Rotation, "grad_method", "B")
+
+        with qml.tape.JacobianTape() as tape:
+            qml.Rotation(1.0, wires=0)
+            qml.expval(qml.X(0))
+
+        with pytest.raises(ValueError, match="unknown gradient method"):
+            _grad_method(tape, 0)
+
+
+class TestTransformObservable:

```

Comment:
Copy and paste of existing tests: https://github.com/PennyLaneAI/pennylane/blob/ab71001bc55c2945a0945f42131c45d77da3512e/tests/tape/test_cv_param_shift.py#L195